pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -591,3 +592,65 @@ func getContainerResources(container v1.Container, replicas float64) *clustersta
 	req.GPU = req.GPU * int64(replicas)
 	return req
 }
+
+//return value is a string for flexibility of expression
+func (gr *GenericResources) GetGenericItemKindStatus(aw *arbv1.AppWrapperGenericResource, namespace string) (completed bool) {
+	dd := gr.clients.Discovery()
+	apigroups, err := restmapper.GetAPIGroupResources(dd)
+	if err != nil {
+		klog.Errorf("Error getting API resources, err=%#v", err)
+	}
+	restmapper := restmapper.NewDiscoveryRESTMapper(apigroups)
+	_, gvk, err := unstructured.UnstructuredJSONScheme.Decode(aw.GenericTemplate.Raw, nil, nil)
+	if err != nil {
+		klog.Errorf("Decoding error, please check your CR! Aborting handling the resource creation, err:  `%v`", err)
+	}
+	//service and podgroup has no conditions return and ignore
+	//TODO: make it configurable?
+	if gvk.GroupKind().Kind == "Service" || gvk.GroupKind().Kind == "PodGroup" {
+		return false
```

Comment:
This shouldn't be needed after suggested changes in caller function.

Reply:
Also once the above is removed, please add a test (if not existing already) that has `completionstatus` field set to `Completed` for a non-pod creating object such as `Service` to ensure this condition made mistakenly by the user does not crash MCAD or alter the behavior of the AW.

Reply:
Added another test case where incorrect completion status is added to the service, it does not crash and the AW status is `Running`